src/main/java/wellnus/reflection/command/UnlikeCommand.java

```diff
@@ -0,0 +1,185 @@
+     * @param questionList Object that contains the data about questions
+     */
+    public UnlikeCommand(HashMap<String, String> arguments, QuestionList questionList) {
+        super(arguments);
+        this.questionList = questionList;
+        this.favQuestionIndexes = questionList.getDataIndex().get(INDEX_ZERO);
+    }
+
+    /**
+     * Get the command itself.
+     *
+     * @return Command: unlike
+     */
+    @Override
+    protected String getCommandKeyword() {
+        return COMMAND_KEYWORD;
+    }
+
+    /**
+     * Get the name of the feature in which this unlike command is generated.
+     *
+     * @return Feature name: reflect
+     */
+    @Override
+    protected String getFeatureKeyword() {
+        return FEATURE_NAME;
+    }
+
+    /**
+     * Method to ensure that developers add in a command usage.
+     * <p>
+     * For example, for the 'add' command in AtomicHabit package: <br>
+     * "usage: add --name (name of habit)"
+     *
+     * @return String of the proper usage of the habit
+     */
+    @Override
+    public String getCommandUsage() {
+        return COMMAND_USAGE;
+    }
+
+    /**
+     * Method to ensure that developers add in a description for the command.
+     * <p>
+     * For example, for the 'add' command in AtomicHabit package: <br>
+     * "add - add a habit to your list"
+     *
+     * @return String of the description of what the command does
+     */
+    @Override
+    public String getCommandDescription() {
+        return COMMAND_DESCRIPTION;
+    }
+
+    /**
+     * Validate the command.<br/>
+     * <br/>
+     * Conditions for command to be valid:<br/>
+     * <li>Only one argument-payload pair
+     * <li>The pair contains key: like
+     * <li>Payload must be string which parse into integer ranges from 1 to 5
+     * Whichever mismatch will cause the command to be invalid.
+     *
+     * @param commandMap Argument-Payload map generated by CommandParser
+     * @throws BadCommandException If an invalid command is given
+     */
+    @Override
+    public void validateCommand(HashMap<String, String> commandMap) throws BadCommandException, NumberFormatException {
+        if (commandMap.size() != ARGUMENT_PAYLOAD_SIZE) {
+            throw new BadCommandException(INVALID_COMMAND_MSG);
+        } else if (!commandMap.containsKey(COMMAND_KEYWORD)) {
+            throw new BadCommandException(INVALID_COMMAND_MSG);
+        } else {
+            int questionIndex = Integer.parseInt(commandMap.get(COMMAND_KEYWORD));
+            if (this.favQuestionIndexes.size() == EMPTY_LIST) {
+                throw new BadCommandException(EMPTY_FAV_LIST_MSG);
+            }
+            if (questionIndex > this.favQuestionIndexes.size() || questionIndex < LOWER_BOUND) {
+                throw new BadCommandException(WRONG_INDEX_MSG);
+            }
+        }
+        assert getArguments().containsKey(COMMAND_KEYWORD) : COMMAND_KEYWORD_ASSERTION;
```

Comment:
I think this is unnecessary because you already catch it as an exception, which means by the time the code reaches this line the assertion is always true. This particular condition is more suitable as an exception also since it's a user error that we want to print on the screen, rather than a developer error